### ISSUE TITLE
applicationautoscaling-controller generates UUID for targeted resource

### ISSUE THREAD

**Issue by tonyksong** [aws-controllers-k8s/community#2110]
**Describe the bug**
Either the application-autoscaler ACK or the Sagemaker API is assigning a random UUID to a targeted resource, resulting in my IAM role not being authorized to perform an action (`sagemaker:DescribeEndpointConfig`) on my specific named resource.

**Steps to reproduce**
Set up IAM policy for a role that allows the `sagemaker:DescribeEndpointConfig` action on a specific named resource. When the role tries to perform the `sagemaker:DescribeEndpointConfig` action, I get an error which I find in CloudTrail:
```
User: arn:aws:sts::4006xxxx5920:assumed-role/gaip-sagemaker-auto-scaling/kiam-kiam is not authorized to perform:
sagemaker:DescribeEndpointConfig on resource: arn:aws:sagemaker:us-east-1:4006xxxx5920:endpoint-config/7f4e1944-1817
4891-9bef-e055a90787a4 because no identity-based policy allows the sagemaker:DescribeEndpointConfig action
```
It looks like either the ACK or Sagemaker API is assigning a random UUID to the endpoint-config resource, and my policy cannot find our specific named resource, since we will not know this UUID in advance, nor are we able to use a wildcard (company policy). Is there a reason why this UUID is getting assigned instead of the actual resource? If I cannot use a wildcard to name my resource that will use this action, how would I be able to grant permission for our endpoint to perform sagemaker:DescribeEndpointConfig?

**Expected outcome**
The targeted resource should have the correct name and not a random UUID, so that my IAM role will be authorized to perform the action on the resource.

**Environment**

* Kubernetes version: Server Version: v1.25.16
* Using EKS (yes/no), if so version? No
* AWS service targeted (S3, RDS, etc.): Sagemaker, CloudWatch
* applicationautoscaling-controller version: 1.0.14
* ArgoCD
* Helm

**Comment by a-hilaly** [comment]
Looking at the sagemaker code, it doesn't look like the we're generating any sort of UUID, so i'm suspecting that this could be a Sagemaker CP behaviour. @aws-controllers-k8s/sagemaker-maintainer any thoughts on this?

**Comment by ananth102** [comment]
How is the endpoint configuration being created in this case? For the endpoint configuration arn, the resource identifier should be the same as the endpoint configuration name.

**Comment by tonyksong** [comment]
I worked with AWS Support, and they were able to pinpoint the issue to the Application Auto Scaling service itself. Currently, the service uses a permissions validation check that passes a randomly generated endpoint config name to the DescribeEndpointConfig call. As this validation check does not acount for IAM policies scoping down to the resource name, it is the cause of the issue. 

AWS reported that the internal team is working to fix this, but I was told the change would not be applied until the beginning of August at the earliest. Since this issue has impact on my company's business, I hope the change goes in soon. Closing this issue - thank you, all.
